broken intern-server tests when actually run with CDN.

This test is wrong: https://github.com/mozilla/fxa-content-server/blob/master/tests/server/routes.js#L234-L255, in particular https://github.com/mozilla/fxa-content-server/blob/master/tests/server/routes.js#L245. 

Node.js Response.headers always uses lower-case headers, but that compares on 'Access-Control-Allow-Origin', so `hasCORSHeaders` is always `false`.

However, if I flip that to compare on 'access-control-allow-origin', then some other tests now have `hasCORSHeaders` as `true`.

@shane-tomlinson - I'll leave it to you to sort out the correct behaviour in these tests.

I see, I see. 

I have applied this patch to the tests:

```diff
diff --git a/tests/server/routes.js b/tests/server/routes.js
index 238cb36..e81a2c8 100644
--- a/tests/server/routes.js
+++ b/tests/server/routes.js
@@ -242,11 +243,16 @@ define([
           assert.equal(res.statusCode, 200);
 
           var headers = res.headers;
-          var hasCORSHeaders = headers.hasOwnProperty('Access-Control-Allow-Origin');
+          var hasCORSHeaders =
+            // Node responds with Access-Control-Allow-Origin,
+            // nginx responds with access-control-allow-origin
+            headers.hasOwnProperty('Access-Control-Allow-Origin') ||
+            headers.hasOwnProperty('access-control-allow-origin');
+
           if (doesURLRequireCORS(url)) {
-            assert.ok(hasCORSHeaders, url);
+            assert.ok(hasCORSHeaders, url + ' should have CORS headers');
           } else {
-            assert.notOk(hasCORSHeaders, url);
+            assert.notOk(hasCORSHeaders, url + ' should not have CORS headers');
           }
         });
```

This introduces a failure:

```
AssertionError: https://latest.dev.lcip.org/styles/44371fd4.en.css should not have CORS headers: expected true to be falsy
  at Assertion.<anonymous>  <node_modules/intern/browser_modules/chai/chai.js:547:10>
  at Assertion.addProperty  <node_modules/intern/browser_modules/chai/chai.js:4240:29>
  at Function.assert.isNotOk  <node_modules/intern/browser_modules/chai/chai.js:2256:35>
  at <tests/server/routes.js:255:20>
  at <node_modules/intern/node_modules/dojo/Promise.ts:393:15>
  at runCallbacks  <node_modules/intern/node_modules/dojo/Promise.ts:11:11>
  at <node_modules/intern/node_modules/dojo/Promise.ts:317:4>
  at run  <node_modules/intern/node_modules/dojo/Promise.ts:237:7>
  at <node_modules/intern/node_modules/dojo/nextTick.ts:44:3>
  at nextTickCallbackWith0Args  <node.js:419:9>
PASS: front end routes - #http get http://latest.dev.lcip.org/legal/privacy (758ms)
FAIL: front end routes - #https get https://latest.dev.lcip.org/legal/terms (465ms)
AssertionError: https://latest.dev.lcip.org/styles/44371fd4.en.css should not have CORS headers: expected true to be falsy
  at Assertion.<anonymous>  <node_modules/intern/browser_modules/chai/chai.js:547:10>
  at Assertion.addProperty  <node_modules/intern/browser_modules/chai/chai.js:4240:29>
  at Function.assert.isNotOk  <node_modules/intern/browser_modules/chai/chai.js:2256:35>
  at <tests/server/routes.js:255:20>
  at <node_modules/intern/node_modules/dojo/Promise.ts:393:15>
  at runCallbacks  <node_modules/intern/node_modules/dojo/Promise.ts:11:11>
  at <node_modules/intern/node_modules/dojo/Promise.ts:317:4>
  at run  <node_modules/intern/node_modules/dojo/Promise.ts:237:7>
  at <node_modules/intern/node_modules/dojo/nextTick.ts:44:3>
  at nextTickCallbackWith0Args  <node.js:419:9>
```

The URL that's failing is: `https://latest.dev.lcip.org/styles/44371fd4.en.css` The test ensures CORS headers are only set for resources served from the CDN, that URL points not to the CDN but to the original server. Digging in.

--- a/server/lib/routes/get-terms-privacy.js
+++ b/server/lib/routes/get-terms-privacy.js
@@ -28,6 +28,7 @@ var templates = require('../legal-templates');
 module.exports = function verRoute (i18n) {
   var DEFAULT_LANG = config.get('i18n.defaultLang');
   var DEFAULT_LEGAL_LANG = config.get('i18n.defaultLegalLang');
+  var STATIC_RESOURCE_URL = config.get('static_resource_url');
 
   var getTemplate = templates(i18n, PAGE_TEMPLATE_DIRECTORY);
 
@@ -63,7 +64,11 @@ module.exports = function verRoute (i18n) {
 
         res.format({
           'text/html': function () {
-            var context = {};
+            var context = {
+              // Note that staticResourceUrl is added to templates as a
+              // build step
+              staticResourceUrl: STATIC_RESOURCE_URL
+            };
             context[page] = template;
 
             // the HTML page removes the header to allow embedding.

--- a/tests/server/routes.js
+++ b/tests/server/routes.js
@@ -242,11 +242,16 @@ define([
           assert.equal(res.statusCode, 200);
 
           var headers = res.headers;
-          var hasCORSHeaders = headers.hasOwnProperty('Access-Control-Allow-Origin');
+          var hasCORSHeaders =
+            // Node responds with Access-Control-Allow-Origin,
+            // nginx responds with access-control-allow-origin
+            headers.hasOwnProperty('Access-Control-Allow-Origin') ||
+            headers.hasOwnProperty('access-control-allow-origin');
+
           if (doesURLRequireCORS(url)) {
-            assert.ok(hasCORSHeaders, url);
+            assert.ok(hasCORSHeaders, url + ' should have CORS headers');
           } else {
-            assert.notOk(hasCORSHeaders, url);
+            assert.notOk(hasCORSHeaders, url + ' should not have CORS headers');
           }
         });
 

